perceval/backends/_naive.py

```diff
@@ -39,6 +39,10 @@ class NaiveBackend(AProbAmpliBackend):
     """Naive algorithm, no clever calculation path, does not cache anything,
        recompute all states on the fly"""
 
+    def __init__(self, permanentor = lambda M : xq.permanent_cx(M, n_threads=1)):
```

Comment:
I'm not a fan of letting any Naive user inject a way to compute permanent in the main interface of the class.

I'd suggest a simpler approach: a `_compute_permanent()` method which is overriden by the approximate naive back-end